### PR TITLE
[ArtemisNG] Docker Orchestrator

### PR DESCRIPTION
**Pull request by mbounabi**
:warning: This is just the first step in a big evolution, comments welcome if you can understand the full scope of the evolution (more tickets to come linked to https://jira.kisio.org/browse/NAVP-1108). Please refrain your urge to put comments everywhere. :warning: 

- new file docker_orchestrator will pop dockers sequentially and run tests

### CODE REVIEW

File: artemis/docker_orchestrator.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    """
+    upCommand = "TAG=dev docker-compose -f docker-compose.yml -f kirin/docker-compose_kirin.yml up -d --remove-orphans"
+    subprocess.Popen(upCommand, shell=True)
+    wait_for_cities_db()
+
+
+def launch_coverages():
+    if not config['DOCKER_COMPOSE_PATH']:
+        raise Exception("DOCKER_COMPOSE_PATH needs to be set")
+    instances_path = config['DOCKER_COMPOSE_PATH'] + "artemis/"
```
Comment (xlqian) [inline artemis/docker_orchestrator.py]: prefer use os.path.join when building filesystem path, it makes your code more portable

File: artemis/docker_orchestrator.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+from artemis.configuration_manager import config
+
+
+def get_containers_list():
+    return docker.DockerClient(version='auto').containers.list()
+
+
+@retry(stop_max_delay=3000000, wait_fixed=2000)
+def wait_for_jormun():
+    query = 'http://localhost:9191/v1/status'
```
Comment (xlqian) [inline artemis/docker_orchestrator.py]: not very important, i think the port(9191) should be configurable one day

File: artemis/docker_orchestrator.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        if len(containers) > 0:
+            raise Exception("Kraken still running...")
+
+    wait_for_kraken_stop()
+
+
+def init_dockers():
+    """
+    Run docker containers with no instance
+    Create 'jormungandr' and 'cities' db
```
Comment (xlqian) [inline artemis/docker_orchestrator.py]: ... and kirin (and elasticksearch, valhalla, bragi etc....), may be the components list should be configurable... again it's not very important... we can do it later

File: artemis/docker_orchestrator.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+
+def wait_for_docker_stop(kraken_name):
+    print("WAITING TO STOP {}".format(kraken_name))
+
+    @retry(stop_max_delay=3000000, wait_fixed=2000)
+    def wait_for_kraken_stop():
+        docker_list = get_containers_list()
+        containers = [x for x in docker_list if kraken_name in x.name]
+        if len(containers) > 0:
```
Comment (xlqian) [inline artemis/docker_orchestrator.py]: if next((x for x in docker_list if kraken_name in x.name), None) :

**Comment by xlqian** [comment]
gg

File: artemis/docker_orchestrator.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    downCommand = "TAG=dev docker-compose -f docker-compose.yml -f kirin/docker-compose_kirin.yml down -v --remove-orphans"
+    subprocess.Popen(downCommand, shell=True)
+
+    wait_for_containers_stop()
+
+
+"""
+MAIN
+"""
+
```
[the remark this answers is not in the document]
Comment (mbounabi) [reply artemis/docker_orchestrator.py]: The file isn't imported for now but, of course! :smiley:

**Pull request merged**